packages/deployer/subtasks/deploy-router.js

```diff
@@ -1,15 +1,22 @@
+const path = require('path');
 const logger = require('../utils/logger');
 const { subtask } = require('hardhat/config');
 const { TASK_COMPILE } = require('hardhat/builtin-tasks/task-names');
+const { getRouterName } = require('../utils/router');
+const relativePath = require('../utils/relative-path');
 
 const { SUBTASK_DEPLOY_CONTRACT, SUBTASK_DEPLOY_ROUTER } = require('../task-names');
 
-subtask(SUBTASK_DEPLOY_ROUTER).setAction(async (_, hre) => {
+subtask(SUBTASK_DEPLOY_ROUTER).setAction(async ({ instance }, hre) => {
   logger.subtitle('Deploying router');
 
   await hre.run(TASK_COMPILE, { force: false, quiet: true });
 
-  const contractPath = hre.deployer.paths.router;
+  const contractPath = path.join(
+    relativePath(hre.config.paths.sources),
```

Comment:
the relativePath should receive hre.config.paths.root as second parameter, because is defaulted to process.cwd but that's not always the case on hardhat projects (because the hardhats project root can be configured to any other path)

Reply:
Done!